10_rest/mohitnag/pkg/rest/rest_test.go

```diff
@@ -0,0 +1,252 @@
+package rest
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/anz-bank/go-course/10_rest/mohitnag/pkg/puppy"
+	"github.com/anz-bank/go-course/10_rest/mohitnag/pkg/puppy/store"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type restSuite struct {
+	suite.Suite
+	handler    *Handler
+	makeStorer func() puppy.Storer
+}
+
+func (s *restSuite) SetupTest() {
+	s.handler = NewRestHandler(s.makeStorer()).(*Handler)
+	err := initialisePuppyStore(s.handler.storer, "./../puppy/store/testdata/many-puppies.json")
+	if err != nil {
+		s.FailNow("Error in test setup")
+	}
+}
+
+func TestStorers(t *testing.T) {
+	suite.Run(t, &restSuite{
+		makeStorer: func() puppy.Storer { return store.NewMapStore() },
+	})
+	suite.Run(t, &restSuite{
+		makeStorer: func() puppy.Storer { return store.NewSyncStore() },
+	})
+
+}
+
+func (s *restSuite) TestHandleGetPuppy() {
+	s.T().Run("Get an existing Puppy", func(t *testing.T) {
```

Comment:
Rather than using multiple `T().Run()` calls in one test, just split it into multiple test functions: `TestHandleGetPuppy` and `TestHandleGetPuppyMissing`. This makes each test easier to read and removes a level of indentation.

Same with all the other tests in this file.

`t.Run()` is much more useful in a loop with table-driven tests.

Reply:
Done